include/z64save.h

```diff
@@ -964,12 +964,26 @@ typedef enum IngoRaceState {
 #define EVENTINF_MARATHON_ACTIVE 0x10
 
 // EVENTINF 0x20-0x24
-#define EVENTINF_INDEX_20_21_22_23_24 2
-#define EVENTINF_20_MASK (1 << 0)
-#define EVENTINF_21_MASK (1 << 1)
-#define EVENTINF_22_MASK (1 << 2)
-#define EVENTINF_23_MASK (1 << 3)
-#define EVENTINF_24_MASK (1 << 4)
+#define EVENTINF_INDEX_HAGGLING_TOWNSFOLK 0x2
+#define EVENTINF_HAGGLING_TOWNSFOLK_MESG_0 0x20
+#define EVENTINF_HAGGLING_TOWNSFOLK_MESG_1 0x21
+#define EVENTINF_HAGGLING_TOWNSFOLK_MESG_2 0x22
+#define EVENTINF_HAGGLING_TOWNSFOLK_MESG_3 0x23
+#define EVENTINF_HAGGLING_TOWNSFOLK_MESG_4 0x24
+
+#define EVENTINF_HAGGLING_TOWNSFOLK_MASK                                                                     \
+    (EVENTINF_MASK(EVENTINF_HAGGLING_TOWNSFOLK_MESG_0) | EVENTINF_MASK(EVENTINF_HAGGLING_TOWNSFOLK_MESG_1) | \
+     EVENTINF_MASK(EVENTINF_HAGGLING_TOWNSFOLK_MESG_2) | EVENTINF_MASK(EVENTINF_HAGGLING_TOWNSFOLK_MESG_3) | \
+     EVENTINF_MASK(EVENTINF_HAGGLING_TOWNSFOLK_MESG_4))
+
+#define ENMU_GET_TALK_FLAGS() \
+    gSaveContext.eventInf[EVENTINF_INDEX_HAGGLING_TOWNSFOLK] & EVENTINF_HAGGLING_TOWNSFOLK_MASK
+
+#define ENMU_SET_TALK_FLAGS(talkFlags) \
+    gSaveContext.eventInf[EVENTINF_INDEX_HAGGLING_TOWNSFOLK] |= (talkFlags);
+
+#define ENMU_RESET_TALK_FLAGS() \
+    gSaveContext.eventInf[EVENTINF_INDEX_HAGGLING_TOWNSFOLK] &= ~(EVENTINF_HAGGLING_TOWNSFOLK_MASK);
```

Comment:
should these justbe local en_mu defines? the system name at the front of the macro suggests so

Reply:
dunno but fwiw, there is currently `ENDAIKU_IS_CARPENTER_RESCUED` and similar currently in z64save.h that works on `carpenterType`

Reply:
we can keep going like this for now, but to me they do kinda make sense to me in the actor headers (or C file, idk)

Reply:
I guess I should say that part of the original plan was to move all uses eventInf and similar (except for like flg_set) into z64save so that all access is done through z64save macros, but i'm less confident on doing so anymore.

Reply:
I dont think I care too much either way right now, but if they are going to be grouped all together in here, I would prefer if the naming convention was EVENTINF_THING consistently (replace the flag table name in the front for which ever it corresponds to, ofc)